app/static/js/script.js

```diff
@@ -1,31 +1,38 @@
-$(function() {
-	
+$(document).on('ready', function() {
 	$(document).mouseup(function(e) {
 		var container = $(".custom-menu-content");
 		var button = $('.glyphicon-th');
-
 		// if the target of the click is not the button,
 		// the container or a descendant of the container
 		if (!button.is(e.target) && !container.is(e.target) && container.has(e.target).length === 0) {
 			container.addClass('hidden');
 		}
 	});
-
 	$('.glyphicon-th').click(function(){
 		$('.custom-menu-content').toggleClass('hidden');
 	});
-
-	var apiUrl = "https://api.github.com/repos/fossasia/badgeyay/git/refs/heads/development";
-	$.ajax({url: apiUrl, success: function(result){
-		var version = result['object']['sha'];
-		var versionLink = 'https://github.com/fossasia/badgeyay/tree/'+version;
-		var deployLink = $('.version').attr('href', versionLink).html(version);
-	}});
-
 	$(".menu-options").click(function(){
 		var i = $(this).data("item");
 		$(".placeholder").text(i);
 		$("input[name='img-default']").val(i);
 	});
+	var apiUrl = "https://api.github.com/repos/fossasia/badgeyay/git/refs/heads/development";
 
-});
\ No newline at end of file
+	$.ajax({
+		url: apiUrl,
+		async: true,
+		success: function(result) {
+			if(typeof result.object !== 'undefined' && typeof result.object.sha !== 'undefined') {
+				var version = result['object']['sha'];
+				var versionLink = 'https://github.com/fossasia/badgeyay/tree/'+version;
+				var deployLink = $('.version').attr('href', versionLink).html(version);
+			} else {
+				alert('Sorry, Server did not Respond Correctly. Please Reload Page or.')
```

Comment:
Thanks for doing so.

Would you like to add the following?
- name of the script `script.js`
- what is expected (not correctly is a claim which can not be proven)
- what you get instead using `JSON.stringify()`

So that we get an error message like this:

> script.js: Sorry, Server did not Respond Correctly. You can try to repload the page or report the error here:  
  http://github.com/fossasia/badgeyay/issues We expected {... JSON ...} but received {... JSON ...}.

Reply:
These are my first few contributions to fossasia, so yeah.

Reply:
Welcome :)